ipatests/test_integration/test_adtrust_install.py

```diff
@@ -303,3 +306,582 @@ def test_ipa_user_s4u2self_pac(self):
+        )
+        result = self.master.run_command(
+            [
+                "ipa-adtrust-install",
+                "-a",
+                self.master.config.admin_password,
+                "--netbios-name",
+                netbios_name,
+                "-U",
+            ],
+            raiseonerr=False,
+        )
+        assert msg in result.stdout_text
+        assert result.returncode == 0
+
+    def test_adtrust_install_with_non_ipa_user(self):
+        """
+        Test that ipa-adtrust-install command returns
+        an error when kinit is done as alias
+        i.e root which is not an ipa user.
+        """
+        msg = (
+            'Unrecognized error during check of admin rights: '
+            'root: user not found'
+        )
+        user = 'root'
+        self.master.run_command(
+            ["kinit", "-E", user],
+            stdin_text=self.master.config.admin_password,
+            raiseonerr=False
+        )
+        result = self.master.run_command(
+            ["ipa-adtrust-install", "-A", user,
+             "-a", self.master.config.admin_password,
+             "-U"], raiseonerr=False
+        )
+        assert result.returncode != 0
+        assert msg in result.stderr_text
+
+    def test_adtrust_install_as_regular_ipa_user(self):
+        """
+        This testcase checks that when regular ipa user
+        does kinit and runs the ipa-adtrust-install
+        command, the command is not run and message
+        is displayed on the console.
+        """
+        user = "ipauser1"
+        passwd = "Secret123"
+        try:
+            tasks.create_active_user(
+                self.master,
+                user,
+                password=passwd,
+                first=user,
+                last=user,
+            )
+            tasks.kinit_as_user(self.master, user, passwd)
+            self.master.run_command(["klist", "-l"])
+            result = self.master.run_command(
+                ["ipa-adtrust-install", "-A", user,
+                 "-a", passwd, "-U"], raiseonerr=False
+            )
+            msg = "Must have administrative privileges to " \
+                  "setup AD trusts on server\n"
+            assert msg in result.stderr_text
+            assert result.returncode != 0
+        finally:
+            self.master.run_command(["kdestroy", "-A"])
+            tasks.kinit_admin(self.master)
+
+    def test_adtrust_install_as_non_root_user(self):
+        """
+        This testcase checks that when regular
+        ipa user logins and then runs ipa-adtrust-install
+        command, the command fails to run
+        """
+        user = "ipauser2"
+        pwd = "Secret123"
+        cmd = ["ipa-adtrust-install"]
+        msg = (
+            "Must be root to setup AD trusts on server"
+        )
+        try:
+            tasks.create_active_user(self.master, user, pwd)
+            result = tasks.run_command_as_user(
+                self.master, user, cmd
+            )
```

Comment:
I forgot that run_command_as_user does not support the raiseonerr=False parameter. The kdestroy and kinit in the finally clause are not needed (the next test will handle that part), the resulting code can look like:
```
        try:
            tasks.create_active_user(self.master, user, pwd)
            result = tasks.run_command_as_user(
                self.master, user, cmd
            )
        except CalledProcessError as e:
            assert msg in e.output
            assert e.returncode != 0
        else:
            pytest.fail("Run ipa-adtrust-install as non-root user did not return error")
```